src/basic/chattr-util.c

```diff
@@ -30,6 +30,9 @@ int chattr_full(
 
         assert(dir_fd >= 0 || dir_fd == AT_FDCWD);
 
+        if (mask == 0 && !ret_previous && !ret_final)
+                return 0;
```

Comment:
my thinking was always that we shouldn't bypass calls like this too aggressively if there's good chance this would cause us to suceed in a case we otherwise wouldn't succeed. which would be the case here i guess: if the fd refers to some weird inode type we'd suddenly get success here rather than ENOTTY.

Is there actually some real-life cases where we'd benefit of the shortcut? if not, i'd probably no bother for this reason

Reply:
But, OK. I will drop the second commit.

Reply:
The second commit is replaced with more specific one. PTAL.